src/reactor.rs

```diff
@@ -585,17 +747,29 @@ mod sys {
             self.0
         }
     }
-    fn flags() -> EventFlag {
-        EventFlag::ONESHOT | EventFlag::IN | EventFlag::OUT | EventFlag::RDHUP
+    fn read_flags() -> EventFlag {
+        EventFlag::IN | EventFlag::RDHUP
+    }
+    fn write_flags() -> EventFlag {
+        EventFlag::OUT
     }
 
     pub struct Events(wepoll_binding::Events);
     impl Events {
         pub fn new() -> Events {
             Events(wepoll_binding::Events::with_capacity(1000))
         }
-        pub fn iter(&self) -> impl Iterator<Item = usize> + '_ {
-            self.0.iter().map(|ev| ev.data() as usize)
+        pub fn iter(&self) -> impl Iterator<Item = Event> + '_ {
+            self.0.iter().map(|ev| Event {
+                readable: ev.flags().intersects(read_flags()),
+                writable: ev.flags().intersects(write_flags()),
```

Comment:
I don't think this will work because wepoll-binding is not reporting back the specific events:

https://gitlab.com/yorickpeterse/wepoll-binding/-/blob/master/src/lib.rs#L132

Reply:
Thank you! This is resolved now.